lib/inferno/dsl/fhir_resource_validation.rb

```diff
@@ -278,6 +281,8 @@ def call_validator(resource, profile_url)
 
         # @private
         def operation_outcome_from_hl7_wrapped_response(response_hash)
+          response_hash = JSON.parse(response_hash) if response_hash.is_a? String
```

Comment:
This really needs a comment saying why this line is needed.  Clearly it is an unexpected workaround to someone that passes a string instead of a hash into this method.

We are also bypassing all the sanitation stuff and invalid json rescuing by letting this work this way. Not sure if you want to fix that here too.  Maybe at least pass it through `remove_invalid_characters()` and just accept invalid json maybe raising an unhelpful purple error for the tests that use this improperly?

Reply:
One thing we could do is not call `#operation_outcome_from_hl7_wrapped_response` ([see here for example](https://github.com/inferno-framework/us-core-test-kit/blob/main/lib/us_core_test_kit/reference_resolution_test.rb#L169)) directly, but call `#operation_outcome_from_validator_response` instead, which runs the response through `#remove_invalid_characters` and `JSON.parse()` before passing it along to `#operation_outcome_from_hl7_wrapped_response`.

Other test kits call `#operation_outcome_from_hl7_wrapped_response` directly from when they were migrated to the new validator. With that the invalid characters method from Vanessa probably goes underutilized and you can run into this string issue.

Reply:
> One thing we could do is not call...

We're fixing it here so that we only have fix it once rather than fixing it in every test kit that makes this call. Making changes in those test kits will be part of the follow-up where we update the DSL to support the actual functionality those tests need.

> Maybe at least pass it through remove_invalid_characters() and just accept invalid json maybe raising an unhelpful purple error for the tests that use this improperly?

Yeah, I'd like to make this change as small as possible, so I think performing the sanitation but not dealing with invalid json right now is fine.

Reply:
I think this is fine for now.  I also think that this really highlights the dangers of having a somewhat fuzzy API into inferno_core/framework functionality now that the team is growing.  We need a flexible and easy to use API for test writers that can do things like this without knowing about the internal functions of core.  Not a problem for this ticket, but I think its a good time to emphasize why this is an important goal for framework.